### PR TITLE
ARC-671: retry loading commits on error "The changedFiles count for this commit is unavailable"

### PR DESCRIPTION
**Pull request by thombergs**
We're seeing a bunch of these errors in the logs, causing backfills to fail. So, we'll catch this error and retry again without asking for the `changedFiles` field.

The `changedFiles` feels can be undefined, it's only used in 2 places and those have a null-check.

### CODE REVIEW

File: src/sync/commits.ts [hunk 1]
```diff
@@ -13,15 +17,44 @@ export default async (github: GitHubAPI, repository, cursor, perPage: number) =>
...
+		}
+
+		// According to the logs, GraphQL queries sometimes fail because the "changedFiles" field is not available.
+		// In this case we just try again, but without asking for the changedFiles field.
+
+		logger.info("retrying without changedFiles");
+
+		const changedFilesErrors = err.errors?.filter(e => e.message?.includes("The changedFiles count for this commit is unavailable"));
+		if (changedFilesErrors.length) {
+			commitsData = (await github.graphql(getCommitsQuery(false), {
```
Comment (rachellerathbone) [inline src/sync/commits.ts:46]: Is there a way to pull this out into a function so we're not repeating the same logic?

File: src/sync/queries.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export const getCommits = `query ($owner: String!, $repo: String!, $per_page: In
                   message
                   oid
                   url
-                  changedFiles
+                  ${includeChangedFiles ? "changedFiles" : ""}
```
Comment (rachellerathbone) [inline src/sync/queries.ts:55]: Has this been tested? Wouldn't graphql shit itself if it tried to query target."" ?

**Review by rachellerathbone — CHANGES_REQUESTED** [review]

File: src/sync/commits.ts (hunk 1 above)
Comment (thombergs) [reply src/sync/commits.ts]: done

File: src/sync/queries.ts (hunk 2 above)
Comment (thombergs) [reply src/sync/queries.ts:55]: GraphQL doesn't care about empty lines. Tested with the GraphQL explorer and with a local sync job

**Review by rachellerathbone — APPROVED** [review]

**Pull request merged**